embassy-rp/src/clocks.rs

```diff
@@ -189,14 +633,147 @@ unsafe fn configure_pll(p: pac::pll::Pll, refdiv: u32, vco_freq: u32, post_div1:
+                $pin_num
+            }
+        }
+    };
+}
+
+impl_gpinpin!(PIN_20, 20, 0);
+impl_gpinpin!(PIN_22, 22, 1);
+
+pub struct Gpin<'d, T: GpinPin> {
+    gpout: PeripheralRef<'d, T>,
+}
+
+impl<'d, T: GpinPin> Gpin<'d, T> {
+    pub fn new(gpout: impl Peripheral<P = T> + 'd) -> Self {
+        into_ref!(gpout);
+
+        unsafe {
+            let p = pac::IO_BANK0.gpio(gpout.pin_number()).ctrl();
+            p.write(|w| w.set_funcsel(0x08));
+        }
+
+        Self { gpout }
+    }
+}
+
+impl<'d, T: GpinPin> Drop for Gpin<'d, T> {
+    fn drop(&mut self) {
+        unsafe {
+            let p = pac::IO_BANK0.gpio(self.gpout.pin_number()).ctrl();
+            p.write(|w| w.set_funcsel(pac::io::vals::Gpio0ctrlFuncsel::NULL.0));
+        }
+    }
+}
+
+pub trait GpoutPin {
+    fn gpout_number(&self) -> usize;
+    fn pin_number(&self) -> usize;
+}
+
+macro_rules! impl_gpoutpin {
+    ($name:ident, $pin_num:expr, $gpout_num:expr) => {
+        impl GpoutPin for crate::peripherals::$name {
+            fn gpout_number(&self) -> usize {
+                $gpout_num
+            }
+            fn pin_number(&self) -> usize {
+                $pin_num
+            }
+        }
+    };
+}
+
+impl_gpoutpin!(PIN_21, 21, 0);
+impl_gpoutpin!(PIN_23, 23, 1);
+impl_gpoutpin!(PIN_24, 24, 2);
+impl_gpoutpin!(PIN_25, 25, 3);
+
+pub struct Gpout<'d, T: GpoutPin> {
+    gpout: PeripheralRef<'d, T>,
+}
+
+impl<'d, T: GpoutPin> Gpout<'d, T> {
+    pub fn new(gpout: impl Peripheral<P = T> + 'd) -> Self {
+        into_ref!(gpout);
+
+        unsafe {
+            let p = pac::IO_BANK0.gpio(gpout.pin_number()).ctrl();
+            p.write(|w| w.set_funcsel(0x08));
+        }
+
+        Self { gpout }
+    }
+
+    pub fn set_div(&self, int: u32, frac: u8) {
+        unsafe {
+            let c = pac::CLOCKS;
+            c.clk_gpout_div(self.gpout.gpout_number()).write(|w| {
+                w.set_int(int);
+                w.set_frac(frac);
+            });
+        }
+    }
+
+    pub fn set_src(&self, src: ClkGpoutCtrlAuxsrc) {
+        unsafe {
+            let c = pac::CLOCKS;
+            c.clk_gpout_ctrl(self.gpout.gpout_number()).modify(|w| {
+                w.set_auxsrc(src);
+            });
+        }
+    }
+
+    pub fn enable(&self) {
+        unsafe {
+            let c = pac::CLOCKS;
+            c.clk_gpout_ctrl(self.gpout.gpout_number()).modify(|w| {
+                w.set_enable(true);
+            });
+        }
+    }
+
+    pub fn disable(&self) {
+        unsafe {
+            let c = pac::CLOCKS;
+            c.clk_gpout_ctrl(self.gpout.gpout_number()).modify(|w| {
+                w.set_enable(false);
+            });
+        }
+    }
+}
+
+impl<'d, T: GpoutPin> Drop for Gpout<'d, T> {
+    fn drop(&mut self) {
+        unsafe {
+            let p = pac::IO_BANK0.gpio(self.gpout.pin_number()).ctrl();
+            p.write(|w| w.set_funcsel(pac::io::vals::Gpio0ctrlFuncsel::NULL.0));
```

Comment:
this could also disable it in `clk_gpout_ctrl`. Not sure if there's any advantage (powr consumption maybe?), but I'd do it just to be safe.